### ISSUE TITLE
RepeatForever and DelayTime

### ISSUE THREAD

**Issue by enrevol** [cocos2d/cocos2d-x#18003]
- cocos2d-x version: latest master branch (41ef696)
- devices test on: iphone simulator
- developing environments
   - Xcode version: 8.2

Steps to Reproduce:

```
some_node->runAction(cocos2d::RepeatForever::create(cocos2d::Sequence::create(
    cocos2d::CallFunc::create([] { CCLOG("AAA"); }),
    cocos2d::DelayTime::create(1.0f), nullptr)));
```
Problem:
- "AAA" is logged once.

Some investigation:
```
bool Sequence::isDone() const
{
    // fix issue #17884
    return (ActionInterval::isDone() && _actions[1]->isDone());
}
```
- `_actions[1]->isDone()` always returns false (`_actions[1]` is `DelayTime`).

**Comment by Bilalmirza** [comment]
@enrevol there has been too many issues since v3.15 on sequences. Which also made me stick with v3.14 ... There are other issues reported as well.

**Comment by minggo** [comment]
@Bilalmirza yep, i created v3.15.2 to fix some critical bugs, include sequence issue.

**Comment by minggo** [comment]
@Bilalmirza v3.15 doesn't modify Actions related codes. The issues only exists in github latest codes.

**Comment by minggo** [comment]
fixed in https://github.com/cocos2d/cocos2d-x/pull/18009.

**Comment by Bilalmirza** [comment]
@minggo I am pretty sure there is a problem v3.15  (not git hub). What i do in  my game is use 

bullet->runAction(Sequence::create(MoveBy, RemoveSelf::create());

what happens is most of the time this works... but after i play a while i keep seeing undeleted bullets on my screen... this only happens when i update the code to v3.15

I hope its fixed in #18009 I will test this when v3.16 comes out

Thanks @minggo for all the help :)

**Comment by minggo** [comment]
@Bilalmirza i think it is fixed in ##17557, but it brings some issues. #18009 is just one PR to fixed issues brought by #17557.

Yep, you can try latest codes in github to have a try.

**Comment by Bilalmirza** [comment]
@minggo thanks again man... ill try the github code and let you know :)
